src/plugins/discover/public/saved_searches/saved_search_alias_match_redirect.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { getSavedSearchUrl } from './saved_searches_utils';
+import { SAVED_SEARCH_TYPE } from './constants';
+
+import type { SavedSearch } from './types';
+import type { SpacesApi } from '../../../../../x-pack/plugins/spaces/public';
+
+interface SavedSearchAliasMatchRedirectProps {
+  savedSearch?: SavedSearch;
+  spaces?: SpacesApi;
+}
+
+export const useSavedSearchAliasMatchRedirect = ({
+  savedSearch,
+  spaces,
+}: SavedSearchAliasMatchRedirectProps) => {
+  useEffect(() => {
+    async function aliasMatchRedirect() {
+      if (savedSearch) {
+        const { aliasTargetId, outcome } = savedSearch.sharingSavedObject ?? {};
+
+        if (spaces && aliasTargetId && outcome === 'aliasMatch') {
+          await spaces.ui.redirectLegacyUrl(
```

Comment:
Small thing: I tested the case of a legacy alias URL combined with some state in the URL - in this case discover will redirect to the actual id and discard any changes of the URL state. I'm not sure whether this is wrong or right, but in Visualize it respects the URL state and uses it over the saved object state. @kertal @timroes do you think the current behavior is fine?

Reply:
ok, state params were added. I don't have strong opinion is it needed or not, but let's do it like in our guide